grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -495,13 +495,11 @@ class Order implements Serializable {
     }
 
     Boolean hasInvoiceableOrderItem() {
-        return orderItems.any { it.invoiceable }
+        return orderItems.any { it.canBeOnRegularInvoice() && it.invoiceable }
```

Comment:
is there a reason to not move the `canBeOnRegularInvoice` methods to within the `invoiceable` methods? Do we ever need to know if an item/adjustment is invoiceable outside of the context of generating a regular invoice for it?

Reply:
It has to be separate for checking if the order `isFullyInvoiceble`. The `canBeOnRegularInvoice` still could be part of `invoiceable` for this case that you pointed out, but since I needed it separately for another place, I decided not to combine these together (at least for now).